crates/compiler/parse/src/expr.rs

```diff
@@ -3245,16 +3271,28 @@ fn record_help<'a>() -> impl Parser<'a, RecordHelp<'a>, ERecord<'a>> {
         reset_min_indent(record!(RecordHelp {
             // You can optionally have an identifier followed by an '&' to
             // make this a record update, e.g. { Foo.user & username: "blah" }.
-            update: optional(backtrackable(skip_second(
-                spaces_around(
-                    // We wrap the ident in an Expr here,
-                    // so that we have a Spaceable value to work with,
-                    // and then in canonicalization verify that it's an Expr::Var
-                    // (and not e.g. an `Expr::Access`) and extract its string.
-                    loc(record_updateable_identifier()),
+            prefix: optional(map_with_arena(
+                either(
+                    backtrackable(skip_second(
+                        spaces_around(
+                            // We wrap the ident in an Expr here,
+                            // so that we have a Spaceable value to work with,
+                            // and then in canonicalization verify that it's an Expr::Var
+                            // (and not e.g. an `Expr::Access`) and extract its string.
+                            loc(record_updateable_identifier()),
+                        ),
+                        byte(b'&', ERecord::Ampersand)
+                    )),
+                    backtrackable(skip_second(
+                        spaces_around(loc(record_builder_mapper_identifier()),),
+                        two_bytes(b'<', b'-', ERecord::Arrow),
+                    )),
                 ),
-                byte(b'&', ERecord::Ampersand)
-            ))),
+                |_arena, output| match output {
+                    Either::First(update) => RecordHelpPrefix::Update(update),
+                    Either::Second(mapper) => RecordHelpPrefix::Mapper(mapper),
+                }
+            )),
```

Comment:
As written, we always parse the mapper identifier twice. That's because it'd first try the update branch and backtrack when it finds the `<-` symbol.

You can improve this by parsing the identifier first, and then parsing either the `&` or `<-` symbols.